native/src/modules/layout/containers/HeaderContainer.js

```diff
@@ -23,21 +24,50 @@ type StatePropsType = {|
   goToLanguageChange?: () => void,
   peeking: boolean,
   categoriesAvailable: boolean,
-  routeCityModel?: CityModel
+  routeCityModel?: CityModel,
+  shareUrl: string | null
 |}
 
 type DispatchPropsType = {| dispatch: Dispatch<StoreActionType> |}
 
 type PropsType = {| ...OwnPropsType, ...StatePropsType, ...DispatchPropsType |}
 
+const getShareUrl = (cityContent, route, routeKey, routeCityModel, shareUrlFromScene): string | null => {
+  let routeMappingType = null
+
+  if (cityContent) {
+    if (cityContent.categoriesRouteMapping[routeKey]) { // TODO refactor in #450
+      routeMappingType = CATEGORIES_ROUTE
+    } else if (cityContent.eventsRouteMapping[routeKey]) {
+      routeMappingType = EVENTS_ROUTE
+    } else if (cityContent.newsRouteMapping[routeKey]) {
+      routeMappingType = NEWS_ROUTE
+    }
+  }
+
+  const path = route.path ? route.path : null
+
+  let shareUrl = null
+  if (path) {
+    shareUrl = url(path)
+  } else if (routeMappingType && routeCityModel) { // for overview pages of news, events and pois, information should be moved to routeMappings in redux store TODO in 450
+    shareUrl = cityContentUrl({ cityCode: routeCityModel.name, languageCode: route.language, route: routeMappingType, path: null })
+  } else if (shareUrlFromScene && typeof shareUrlFromScene === 'string') { // for routes not saved in redux store eg. offers
+    shareUrl = shareUrlFromScene
+  }
+
+  return shareUrl
+}
+
```

Comment:
How about:
```suggestion
const getShareUrl = (params: {| city: string, language: string, path: ?string, routeName: string |}): string | null => {
  const { city, language, path, routeName } = params
  if (path) {
    return url(path)
  } else if ([EVENTS_ROUTE, NEWS_ROUTE, POIS_ROUTE].includes(routeName)) {
    const params = { cityCode: city, languageCode: language, path: null }
    return cityContentUrl({ route: routeName, ...params })
  }
  return null
}
```
Together with the suggestion below this should work.